stage0/gc.go

```diff
@@ -65,55 +66,94 @@ type mount struct {
 	id         int
 	parentId   int
 	mountPoint string
+	mode       string
 }
 
 type mounts []*mount
 
-// Less ensures that mounts are sorted in an order we can unmount; child before
-// parent. This works by pushing all child mounts up to the front. Parent
-// mounts are halted once they encounter a child. The actual sequential order
-// of the mounts doesn't matter; only that children are in front of parents.
-func (m mounts) Less(i, j int) bool { return m[i].id != m[j].parentId }
-func (m mounts) Len() int           { return len(m) }
-func (m mounts) Swap(i, j int)      { m[i], m[j] = m[j], m[i] }
+// getMountDepth searches the list for all ancestors of the mount at index i
+func (m mounts) getMountDepth(i int) int {
+	ancestorMap := make(map[int]struct{}, 0)
```

Comment:
you could just do `ancestrorCount := 0`

And below, `ancestrorCount = ancestrorCount + 1`